.github/workflows/golang-release.yml

```diff
@@ -0,0 +1,69 @@
+name: Go Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version_number:
+        description: "New Version"
+        required: true
+
+env:
+  GO111MODULE: on
+  GO_VERSION: "^1.18"
+
+jobs:
+  build_rust:
+    uses: ./.github/workflows/build-rust-cross-platform.yml
+
+  generate-schemas:
+    uses: ./.github/workflows/generate_schemas.yml
+
+  build:
+    needs: [build_rust, generate-schemas]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
```

Comment:
I am not familiar with using the `restore-keys` with this action, will this name be sufficient? Or is there a glob pattern expected here? Ex. `${{ runner.os }}-go-*`?

Reply:
This should work, if you look at [this](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key)